slam-libraries/viam-orb-slam3/orbslam_server_v1.h

```diff
@@ -30,6 +35,10 @@ namespace viam {
 static const int filenamePrefixLength = 6;
 static const int checkForShutdownIntervalMicroseconds = 1e5;
 extern std::atomic<bool> b_continue_session;
+// Byte limit on GRPC, used to help determine sampling skip_count
+static const int maximumGRPCByteLimit = 32 * 1024 * 1024;
+// Byte limit for chunks on GRPC, used for streaming apis
+static const int maximumGRPCByteChunkSize = 64 * 1024;
```

Comment:
Why do we use a smaller value for maximumGRPCByteChunkSize than maximumGRPCByteLimit?

Reply:
The maximumGRPCByteLimit is 32 MB which is quite large.  I recommended smaller ByteChunkSizes of 64 kb so that we minimize memory consumption of the buffered response both on the slam algo side & the RDK caller side.

Reply:
We increased from 4MB to 32MB to allow pointclouds/maps beyond 4MB to be sent. With streaming that reason no longer holds.

However general consensus seems to be: "The 4 MB limit is protect clients/servers who haven't thought about message size constraints. gRPC itself is fine with going much higher (100s of MBs), but most applications could be trivially attacked or accidentally go out-of-memory allowing messages of that size.

If you're willing to receive a 100 MB message all-at-once, then increasing the limit is fine."

We should be able to change the chunk size to 32MB if we are confident that no 3rd party apps, that will need to ingest the streaming data, have data constraints on them.

Reply:
re `If we are using a smaller chunk size, then why do we increase the max message size from 4MB to 32MB? Is this just to be consistent with rdk?` 

I don't see where we are changing the max unary message size from 4MB to 32BM. Where do you see that?

Reply:
The changes in [slam-libraries/viam-orb-slam3/orbslam_server_v1_main.cc](https://github.com/viamrobotics/slam/pull/149/files#diff-3d6d11e825448c5db341a7bc8b68686e579dcf74c77b90b9638fcc05358f5e75)

Can we remove the line to increase the max message size after the streaming changes are done?

Reply:
Sorry I might be missing something. I think John is just moving a magic number into a constant with the same value.

The change is going from:
```
   builder.SetMaxSendMessageSize(32 * 1024 * 1024);
```

to
```
builder.SetMaxSendMessageSize(viam::maximumGRPCByteLimit);
```

where `viam::maximumGRPCByteLimit` is equal to 
```
static const int maximumGRPCByteLimit = 32 * 1024 * 1024;
```

So I don't think this is going to change any behavior at all. Am I still missing something?

Reply:
No, it's not a behavior change. It was just something I noticed as part of the code review.

Reply:
oh yeah, I realized we had a magic number in that spot so I swapped to using the variable that was consistent with the cartographer implementation

Reply:
Got it. Understood. Sorry for my confusion.